### ISSUE TITLE
Define serialization format for behavioral cloning datasets

### ISSUE THREAD

**Issue by cswinter** [entity-neural-network/incubator#5]
Define a serialization format for behavioral cloning datasets of entity-gym observation/action pairs and add helper methods to serialize/deserialize datasets. Probably using [MessagePack](https://github.com/msgpack/msgpack-python).

**Comment by cswinter** [comment]
Partial progress in https://github.com/entity-neural-network/incubator/pull/29
Still missing:
- [ ] serialize `ActionSpace`
- [ ] serialize `ObsSpace`
- [ ] Record full logprobs, not just logprob of selected action
- [ ] Record chosen action
- [ ] Capture returns rather than instantaneous rewards

**Comment by cswinter** [comment]
Some more progress in https://github.com/entity-neural-network/incubator/pull/147, we now record the full logits and chosen actions.
